src/odemis/odemisd/modelgen.py

```diff
@@ -808,7 +839,7 @@ def instantiate_component(self, name):
         # Add to the microscope all the new components that should be child
         mchildren = self._microscope_ast["children"].values()
         # we only care about children created by delegation, but all is fine
-        newcmps = self.get_children(comp) # that includes comp itself
```

Comment:
The goal here is to catch all the newly created children. So, `get_children` should be the right thing to do (ie, don't get the dependencies).